pytorch_lightning/trainer/trainer.py

```diff
@@ -1521,7 +1521,8 @@ def _call_ttp_hook(
         hook_name: str,
         *args: Any,
         **kwargs: Any,
-    ):
+    ) -> Optional[Any]:
```

Comment:
unfortunately this causes a mypy error, which is why I had to remove it in my PR